packages/create/create.js

```diff
@@ -501,10 +513,149 @@ WantedBy=multi-user.target
 	}
 }
 
+async function copyPluginTemplateFile(src, dst, props) {
+	logger.verbose(`Writing ${dst} from template ${src}`);
+	if (await fs.exists(dst)) {
+		logger.warn(`Could not create file ${dst} because it already exists`);
+		return;
+	}
```

Comment:
The check happens before the templates are read and processed, therefore is it not better to check first because it avoids performing work that may be discarded?

Reply:
As highlighted on [discord](https://discord.com/channels/450361298220351489/450361298220351491/1212346123386232842) using `fs.exists` like this does add a race condition even if nothing else is expected to modify the file system. Therefore it has been switched to a try catch.

Reply:
The file existing is an edge case. We do not need to be concerned with performance in this edge case because having the command run faster on the second run when it fails to output anything is irrelevant to the user.